kolibri_instant_schools_plugin/assets/src/views/profile-page/index.vue

```diff
@@ -177,52 +185,41 @@
         return null;
       },
       permissionTypeText() {
-        if (this.permissionType === PermissionTypes.SUPERUSER) {
-          return this.$tr('isSuperuser');
-        } else if (this.permissionType === PermissionTypes.LIMITED_PERMISSIONS) {
-          return this.$tr('limitedPermissions');
+        switch (this.permissionType) {
+          case PermissionTypes.SUPERUSER:
+            return this.$tr('isSuperuser');
+          case PermissionTypes.LIMITED_PERMISSIONS:
+            return this.$tr('limitedPermissions');
+          default:
+            return '';
         }
-        return '';
-      },
-      canEditUsername() {
-        if (this.isCoach || this.isLearner) {
-          return this.facilityConfig.learnerCanEditUsername;
-        }
-        return true;
       },
       canEditName() {
-        if (this.isCoach || this.isLearner) {
-          return this.facilityConfig.learnerCanEditName;
-        }
-        return true;
+        return this.userIsAdmin || this.facilityConfig.learnerCanEditName;
       },
-      nameIsInvalidText() {
-        if (this.nameBlurred || this.formSubmitted) {
-          if (this.name === '') {
-            return this.$tr('required');
-          }
-        }
-        return '';
+      canEditPassword() {
+        return this.userIsAdmin || this.facilityConfig.learnerCanEditPassword;
       },
-      nameIsInvalid() {
-        return !!this.nameIsInvalidText;
+      // i.e. not a coach or a learner
+      userIsAdmin() {
+        return !this.isCoach && !this.isLearner;
       },
-      usernameIsInvalidText() {
-        if (this.usernameBlurred || this.formSubmitted) {
-          if (this.username === '') {
-            return this.$tr('required');
-          }
-          if (!validateUsername(this.username)) {
-            return this.$tr('usernameNotAlphaNumUnderscore');
-          }
-        }
-        return '';
+      nameIsValid() {
+        return this.name.trim() !== '';
```

Comment:
The k-textbox already trims  https://github.com/learningequality/kolibri/blob/develop/kolibri/core/assets/src/views/k-textbox/index.vue#L7